pkg/syncer/spec/mutators/deployment.go

```diff
@@ -64,9 +79,49 @@ func (dm *DeploymentMutator) Mutate(downstreamObj *unstructured.Unstructured) er
 	// Alias for the template.spec to improve readability.
 	templateSpec := &deployment.Spec.Template.Spec
 
+	desiredServiceAccountName := "default"
+	if templateSpec.ServiceAccountName != "" && templateSpec.ServiceAccountName != "default" {
+		desiredServiceAccountName = templateSpec.ServiceAccountName
+	}
+
+	secretList, err := dm.secretLister.ByIndex(dm.indexName, dm.clusterName.String())
+	if err != nil {
+		return fmt.Errorf("error listing secrets for cluster %s: %w", dm.clusterName, err)
+	}
+
+	desiredSecretName := ""
+	for _, secretObj := range secretList {
+		var secret *corev1.Secret
+		switch obj := secretObj.(type) {
```

Comment:
not needed. A lister always returns the right objects.

Reply:
The indexers we have here are dynamic indexers, aren't they? So always unstructured.

Reply:
I get a *corev1.Secret when it comes from tests, that's why I need to handle that case.

Reply:
@jmprusi is there any way to avoid doing that? We'd rather not have special casing main code that has portions that are only there to accommodate test behavior.

Reply:
done